src/main/java/work/xeltica/craft/core/modules/coreProtectGuiTest/CoreProtectGuiTestModule.kt

```diff
@@ -0,0 +1,371 @@
+
+    /**
+     * [CoreProtectGuiTestApp]
+     *
+     * NOTE: アプリとモジュールでのやりとりがかなり多いのでインスタンス生成処理の削減のためにメンバ変数として宣言する
+     */
+    private val app by lazy { CoreProtectGuiTestApp() }
+    private val gui by lazy { Gui.getInstance() }
+
+    override fun onEnable() {
+        // むかつくので消さない
+        Bukkit.getLogger().info("モジュールをボンジュール～")
+    }
+
+    /**
+     * テキストでログを取るプレイヤーを指定する
+     *
+     * @param player コマンドを打つプレイヤー
+     */
+    private fun getInputTextByConsole(player: Player) {
+        gui.openTextInput(player, "プレイヤーのIDを指定してください") { userName ->
+            userName.let {
+                app.onInputPlayerName(it, player)
+            }
+        }
+        return
+    }
+
+    /**
+     * メニューからログを取るプレイヤーを指定する
+     *
+     * @param player コマンドを打つプレイヤー
+     */
+    private fun getChooseByPlayerMenu(player: Player) {
+        gui.openPlayersMenu(player, "プレイヤーを指定してください", { playerName ->
+            playerName.player?.name?.let {
+                app.onInputPlayerName(it, player)
+            }
+        }, { it.uniqueId != player.uniqueId || it.uniqueId == player.uniqueId })
+        return
+    }
+
+    /**
+     * coreProtectの処理で何かをするときに呼ばれる
+     *
+     * @param player 何かを開始するプレイヤー
+     * @return メニューアイテムのリスト
+     */
+    private fun coreProtectModeList(player: Player): List<MenuItem> {
+        return listOf(
+                MenuItem("ログを検索する", { lookupModeStart(player) }, Material.WRITABLE_BOOK),
+                MenuItem("ロールバック", { rollbackModeStart(player) }, Material.ENDER_CHEST),
+                MenuItem("ページを取得する", { inputLookUpPage(player) }, Material.BOOK),
+                MenuItem("インスペクトモード切り替え", { app.onInspectMode(player) }, Material.ENDER_EYE),
+                MenuItem("キャンセル", { app.onCancel(player) }, Material.BARRIER),
+        )
+    }
+
+    /**
+     * ログを検索する
+     *
+     * @param player ログを取りにいくプレイヤー
+     */
+    private fun lookupModeStart(player: Player) {
+        app.commandFirst = "co lookup "
+        showMenu(player, "ログを取るプレイヤーを指定", chooseHowToGetPlayerName(player))
+    }
+
+    /**
+     * ロールバックを実行する
+     *
+     * @param player ロールバックを実行するプレイヤー
+     */
+    private fun rollbackModeStart(player: Player) {
+        app.commandFirst = "co rollback "
+        showMenu(player, "どのプレイヤーの行動をロールバックするのかを指定", chooseHowToGetPlayerName(player))
+    }
+
+    /**
+     * ページ番号を入力させる
+     *
+     * @param player コマンドを打ちたいプレイヤー
+     */
+    private fun inputLookUpPage(player: Player) {
+        gui.openTextInput(player, "閲覧するページを選択してください") { inputString ->
+            val value = inputString.toIntOrNull()
+            value?.let {
+                if (it <= 0) {
+                    Gui.getInstance().error(player, "正しい数値を入力する必要があります。")
+                    return@openTextInput
+                }
+
+                app.commandSend(value, player)
+            } ?: run {
+                Gui.getInstance().error(player, "正しい数値を入力する必要があります。")
+                return@openTextInput
+            }
```

Comment:
`if` のほうが可読性が高いと感じました。
もし `let` & `run` を使うべき理由が特に無いのであれば、以下のよう書くことを提案します。

```suggestion
            if (value == null || value <= 0) {
                Gui.getInstance().error(player, "正しい数値を入力する必要があります。")
                return@openTextInput
            }

            app.commandSend(value, player)
```